plugins/besecure.rb

```diff
@@ -0,0 +1,310 @@
+      print_status("besecure_debug                 Enable/Disable debugging")
+      print_status("besecure_version               Display the version of the beSECURE server")
+      print_status("besecure_apikey                Set the beSECURE API Key")
+      print_status("besecure_hostname              Set the beSECURE Hostname")
+
+      print_status
+      print_status("REPORTS")
+      print_status("=======")
+      print_status("besecure_report_list           Lists reports")
+      print_status("besecure_report_download       Downloads an beSECURE report specified by ID")
+      print_status("besecure_report_import         Import report specified by ID into framework")
+    end
+
+    # Verify the database is connected and usable
+    def database?
+      if !(framework.db and framework.db.usable)
+        return false
+      else
+        return true
+      end
+    end
+
+    # Verify correct number of arguments and verify -h was not given. Return
+    # true if correct number of arguments and help was not requested.
+    def args?(args, min=1, max=nil)
+      if not max then max = min end
+      if (args.length < min or args.length > max or args[0] == "-h")
+        return false
+      end
+
+      return true
+    end
+
+  #--------------------------
+  # Basic Functions
+  #--------------------------
+  def cmd_besecure_hostname(*args)
+    if args?(args)
+      @hostname = args[0]
+      print_good(@hostname)
+    else
+      print_status("Usage:")
+      print_status("besecure_hostname string")
+    end
+  end
+
+  def cmd_besecure_apikey(*args)
+    if args?(args)
+      @apikey = args[0]
+      print_good(@apikey)
+    else
+      print_status("Usage:")
+      print_status("besecure_apikey string")
+    end
+  end
+
+  def cmd_besecure_debug(*args)
+    if args?(args)
+      @debug = args[0].to_i
+      print_good(@debug)
+    else
+      print_status("Usage:")
+      print_status("besecure_debug integer")
+    end
+  end
+
+  def cmd_besecure_version()
+    req = Net::HTTP::Post.new('/json.cgi', initheader={'Host'=>@hostname})
+    req.set_form_data({'apikey' => @apikey, 'primary' => 'interface'})
+
+    if @debug
+      print_status(req.body)
+    end
+      
+    http = Net::HTTP::new(@hostname, 443)
+    http.use_ssl = true
+    res = http.start {|http| http.request(req)}
+
+    unless res
+      print_error("#{@hostname} - Connection timed out")
+      return ''
+    end
+
+    if @debug
+      print_status(res)
+    end
+
+    body = ''
+    begin
+      body = JSON.parse(res.body)
+    rescue JSON::ParserError
+      print_error("#{@hostname} - Unable to parse the response")
+      return ''
+    end
+
+    if body['error']
+      print_error("#{@hostname} - An error occured:")
+      print_error(body)
+      return ''
+    end
+
+    print_good(body['version'])
+  end
+
+  #--------------------------
+  # Report Functions
+  #--------------------------
+  
+    def cmd_besecure_report_list(*args)
```

Comment:
This non-gracefully fails for me if I don't specify host correctly:

```
msf5 > load besecure
msf5 > besecure_report_list
[-] Error while running command besecure_report_list: Failed to open TCP connection to :443 (Connection refused - connect(2) for nil port 443)
```

And if I supply `hostname` but not a key, it's a little cryptic to understand what's got wrong. It might be nice to add a guard clause against these methods :+1:

Reply:
I will amend and reissue a patch